File: src/main/scala/Wiki.scala
import argonaut._
import Argonaut._

case class Wiki(articles : Map[String, WikiArticle],
                categories : Map[String, WikiCategory])

case class WikiArticle(alternatives : Seq[String],
  categories : Seq[String])

case class WikiCategory(categories : Seq[String])

object Wiki {
  implicit def wikiCodecJson : CodecJson[Wiki] =
    casecodec2(Wiki.apply, Wiki.unapply)("articles", "categories")
}

object WikiArticle {
  implicit def codecJson : CodecJson[WikiArticle] =
    casecodec2(WikiArticle.apply, WikiArticle.unapply)("alternatives", "categories")
}

object WikiCategory {
  implicit def codecJson : CodecJson[WikiCategory] =
    casecodec1(WikiCategory.apply, WikiCategory.unapply)("categories")
}

File: annotool-rs/src/wiki.rs
use std::collections::HashMap;

#[derive(Debug,Serialize,Deserialize)]
pub struct Wiki {
    articles : HashMap<String, WikiArticle>,
    categories : HashMap<String, WikiCategory>
}

#[derive(Debug,Serialize,Deserialize)]
pub struct WikiArticle {
    alternatives : Vec<String>,
    categories : Vec<String>
}

#[derive(Debug,Serialize,Deserialize)]
pub struct WikiCategory {
    categories : Vec<String>
}

